packages/angular/ssr/src/app-engine.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { lookup as lookupMimeType } from 'mrmime';
+import { AngularServerApp } from './app';
+import { Hooks } from './hooks';
+import { getPotentialLocaleIdFromUrl } from './i18n';
+import { AngularAppEngineManifest } from './manifest';
+
+/**
+ * Angular server application engine.
+ * Manages localized Angular server applications, handles rendering requests,
+ * and optionally transforms index HTML before rendering.
+ */
+export class AngularAppEngine {
+  /**
+   * Hooks for extending or modifying the behavior of the server application.
+   * @internal This property is accessed by the Angular CLI when running the dev-server.
+   */
+  static hooks = new Hooks();
+
+  /**
+   * Hooks for extending or modifying the behavior of the server application.
+   * This instance can be used to attach custom functionality to various events in the server application lifecycle.
+   */
+  get hooks(): Hooks {
+    return AngularAppEngine.hooks;
+  }
+
+  /**
+   * Indicates whether the application is in development mode.
+   * @internal This property is set by the Angular CLI when running the dev-server.
+   */
+  static isDevMode = false;
```

Comment:
This will primarily be used and configured by the dev-server, regardless of the `ngDevMode` value. 

This is necessary to bypass the internal caching logic, which would otherwise prevent changes from being reflected.